src/panel/app_panel.js

```diff
@@ -1,137 +1,55 @@
-    this.directionPanel = this.directionEnabled ? new ReactPanelWrapper(DirectionPanel) : null;
+    this.initMap();
 
-    this.panels = [
-      this.servicePanel,
-      this.favoritePanel,
-      this.poiPanel,
-    ];
-    if (this.categoryEnabled) {
-      this.panels.push(this.categoryPanel);
-    }
-    if (this.eventEnabled) {
-      this.panels.push(this.eventListPanel);
-    }
-    if (this.directionPanel) {
-      this.panels.push(this.directionPanel);
-    }
+    SearchInput.initSearchInput('#search');
 
-    ReactDOM.render(<Menu />, document.querySelector('.react_menu__container'));
     Telemetry.add(Telemetry.APP_START, null, null, {
       'language': window.getLang(),
       'is_mobile': Device.isMobile(),
     });
-
-    this.initRouter();
     if (performanceEnabled) {
-      window.times.appRendered = Date.now();
-      listen('map_loaded', () => {
-        window.times.mapLoaded = Date.now();
-      });
+      listen('map_loaded', () => { window.times.mapLoaded = Date.now(); });
     }
 
-    const mapHash = parseMapHash(window.location.hash);
-    this.initMap(mapHash);
-  }
-
-  initMap(mapHash) {
-    import(/* webpackChunkName: "map" */ '../adapters/scene')
-      .then(({ default: Scene }) => {
-        this.scene = new Scene();
-        this.scene.initScene(mapHash);
-      });
-  }
-
-  initRouter() {
     this.router = new Router(window.baseUrl);
 
-    this.router.addRoute('Category', '/places/(.*)', placesParams => {
-      window.execOnMapLoaded(() => {
-        this.openCategory(parseQueryString(placesParams));
-      });
-    });
-
-    this.router.addRoute('EventListPanel', '/events/(.*)', eventsParams => {
-      window.execOnMapLoaded(() => {
-        this.openEvents(parseQueryString(eventsParams));
-      });
-    });
-
-    this.router.addRoute('POI', '/place/(.*)', async (poiUrl, options = {}) => {
-      const poiId = poiUrl.split('@')[0];
-      this.openPoi({ ...options, poiId });
-    });
-
-    this.router.addRoute('Favorites', '/favs', () => {
-      this.openFavorite();
-    });
-
-    this.router.addRoute('Routes', '/routes(?:/?)(.*)', (routeParams, options) => {
-      fire('move_mobile_bottom_ui', 0);
-      this.openDirection({ ...parseQueryString(routeParams), ...options });
-    });
-
-    this.router.addRoute('Direct search query', '/([?].*)', queryString => {
-      const params = parseQueryString(queryString);
-      if (params.q) {
-        SearchInput.executeSearch(params.q);
-      } else {
-        this.navigateTo('/');
-      }
-    });
-
-    // Default matching route
-    this.router.addRoute('Services', '/?', (_, options = {}) => {
-      this.resetLayout();
-      if (options.focusSearch) {
-        SearchInput.select();
-      }
-    });
-
     window.onpopstate = ({ state }) => {
       this.router.routeUrl(getCurrentUrl(), state);
     };
 
-    // Route the initial URL
-    this.router.routeUrl(getCurrentUrl());
+    ReactDOM.render(
```

Comment:
Excepting modals, this is now the only place where we mount something explicitely in the DOM.